DNN Platform/HttpModules/DependencyInjection/ServiceRequestScopeModule.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Web;
+using DotNetNuke.Common.Extensions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Web.Infrastructure.DynamicModuleHelper;
+
+[assembly: PreApplicationStartMethod(typeof(DotNetNuke.HttpModules.DependencyInjection.ServiceRequestScopeModule), nameof(DotNetNuke.HttpModules.DependencyInjection.ServiceRequestScopeModule.InitModule))]
+
+namespace DotNetNuke.HttpModules.DependencyInjection
+{
+    public class ServiceRequestScopeModule : IHttpModule
```

Comment:
Does this mean you got Dependency Injection working in HTTP Modules or is this needed to access the System.Web Scope?

Reply:
This HttpModule is the one that handles the entire request scope lifetime.
As stated in the TODOs, this should be the first HttpModule in the pipeline for BeginRequest so that the scope is opened as soon as possible in the request lifetime (all HttpModules after it can access the opened scope through the instance stored in HttpContext.Items - they will not have constructor DI though). The same HttpModule should be the last that executes it's EndRequest so that the scope is disposed as late as possible.

Reply:
Perfect, thanks for the explanation on what is going on with this code. It helps me a lot and I'm sure it will help others that review this change later.

This sounds like the correct way and I would argue the way I should have implemented this in the first place. I'm actually fine keeping this as an attribute on the class. Something like this shouldn't be configured in the web.config, creates another point of failure on a part of the platform that isn't designed to be plug and play